### PR TITLE
cargo-build-bpf: Add Windows support

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

`cargo-build-bpf` doesn't support windows

#### Summary of Changes

Add support for it!

A few notes:
* `ld` doesn't ship by default with mingw, causing `linker not found` errors, so one of the changes is to force the usage of `ld.lld` from the sdk to link.  I have very little knowledge of different linker flavors, but this was the only one that worked.  If you think it should be another one, please let me know.
* it isn't possible to run bash scripts from the program, since they aren't natively executable by windows.  To get around that, this avoids calling `strip.sh` and just calls `llvm-objcopy` directly.
* for `dump.sh`, it's possible to run the script separately, and re-run `cargo build-bpf --dump` to do the postprocessing step.  it isn't elegant, but it actually works!

Fixes #

### CODE REVIEW

**Review by dmakarov — DISMISSED** [review]

**Comment by dmakarov** [comment]
This looks good to me, but I'm not quite sure why the usual CI checks were not executed for this PR. Let's wait until bpf-tools release assets are published ( https://github.com/solana-labs/bpf-tools/actions/runs/1281082328 ) before merging this.

**Comment by joncinque** [comment]
Sounds good to me!  I've nudged a little update to the error message to restart the CI job

**Review by dmakarov — APPROVED** [review]

**Comment by dmakarov** [comment]
Jon, I had to turn off the build on windows when I released bpf-tools v1.19.  The build failed artificats were copied in preparation for making a package tarball. The reason for the failure was the CI agent ran out of disk space.  I'm not sure how to request a bigger disk for a CI agent machine, and so for now CI doesn't build bpf-tools for Windows.

**Comment by joncinque** [comment]
Ah that's too bad.  Thanks for noticing this.  It looks like we'll need to move the steps to a different directory for this to work as expected, if https://github.com/actions/virtual-environments/issues/1341 is actually possible, we can pass in a new `out` directory to checkout and build everything.  I'll take a longer look at this later.

**Review by dmakarov — DISMISSED** [review]

**Comment by gajojr** [comment]
So what is the solution for this?

**Comment by TradeUpCards** [comment]
> So what is the solution for this?

I'd also like to know. Just getting into Solana coding and can't even do Hello World example because it says 

no such subcommand: `+bpf`

Been scouring the web, but seems most people getting it to work are on Linux and not Windows.

**Comment by joncinque** [comment]
Be sure to install the solana command line tools and rust, then use a  bash-compatible shell like Git Bash.  If you need more support, please ask in the Solana Stack Exchange https://solana.stackexchange.com/

**Pull request merged**
